src/parser/shell_expand/words/methods/strings.rs

```diff
@@ -183,7 +183,96 @@ impl<'a> StringMethod<'a> {
                     None
                 };
                 output.push_str(&out.unwrap_or(0).to_string());
-            }
+            },
+            "unescape" => {
+                let word = if let Some(value) = expand.variable(variable, false) {
+                    Some(value)
+                } else if is_expression(variable) {
+                    Some(expand_string(variable, expand, false).join(" "))
+                } else {
+                    None
```

Comment:
Maybe simply return if the value is `None`? Would avoid the need for `Option` here.